qgis/examples/elevation_tile_downloader/plugin.py

```diff
@@ -0,0 +1,73 @@
+# -*- coding: utf-8 -*-
+"""
+/***************************************************************************
+ElevationTileDownloader
+A QGIS plugin
+Downloads elevation tiles
+Generated by Plugin Builder: http://g-sherman.github.io/Qgis-Plugin-Builder/
+                              -------------------
+        begin                : 2021-09-01
+        git sha              : $Format:%H$
+        copyright            : (C) 2021 by Christian Beiwinkel
+        email                : christian@gis-ops.com
+ ***************************************************************************/
+
+/***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+"""
+from pathlib import Path
+from typing import Optional
+
+from qgis.PyQt.QtGui import QIcon
+from qgis.PyQt.QtWidgets import QAction
+from qgis.gui import QgisInterface
+
+from .gui.tile_downloader_dlg import ElevationTileDownloaderDialog
+
+
+class ElevationTileDownloader:
+    """QGIS Plugin Implementation."""
+
+    def __init__(self, iface: QgisInterface):
+        """Constructor."""
+        self.iface = iface
+
+        self.dlg: Optional[ElevationTileDownloaderDialog] = None
+        self.action: Optional[QAction] = None
+
+    def initGui(self):
+        """Create the menu entries and toolbar icons inside the QGIS GUI."""
+        icon = QIcon(str(Path(__file__).parent.joinpath("icon.png")))
+        self.action = QAction(
+            icon, "Download elevation tiles", self.iface.mainWindow()
+        )
+        self.action.triggered.connect(self.run)
+
+        # add action to the raster menu and the default plugin toolbar
+        self.iface.addPluginToRasterMenu(
+            "Elevation Tile Downloader", self.action
+        )
+        self.iface.addToolBarIcon(self.action)
+
+    def unload(self):
+        """Removes the plugin menu item and icon from QGIS GUI."""
+        self.iface.removePluginRasterMenu(
+            self.action.objectName(), self.action
+        )
+        self.iface.removeToolBarIcon(self.action)
+
+    def run(self):
+        """Run method that performs all the real work"""
+
+        # Only create GUI ONCE in callback, so that it will only load when the plugin is started
+        if not self.dlg:
+            self.dlg = ElevationTileDownloaderDialog(self.iface)
```

Comment:
this self.start = True we should get rid of again I think in the plugin builder. pretty sure I PRd that even, but years ago. now I think it's enough to see if self.dlg was already instantiated before, saves a variable